Microsoft.Toolkit/Diagnostics/Internals/ThrowHelper.Guard.cs

```diff
@@ -0,0 +1,219 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+using Microsoft.Toolkit.Extensions;
+
+#nullable enable
+
+namespace Microsoft.Toolkit.Diagnostics
+{
+    /// <summary>
+    /// Helper methods to efficiently throw exceptions.
+    /// </summary>
+    public static partial class ThrowHelper
+    {
+        /// <summary>
+        /// Returns a formatted representation of the input value.
+        /// </summary>
+        /// <param name="obj">The input <see cref="object"/> to format.</param>
+        /// <returns>A formatted representation of <paramref name="obj"/> to display in error messages.</returns>
+        [Pure]
+        private static string ToAssertString(this object? obj)
+        {
+            return obj switch
+            {
+                string _ => $"\"{obj}\"",
+                null => "null",
+                _ => $"<{obj}>"
+            };
+        }
+
+        /// <summary>
+        /// Throws an <see cref="ArgumentException"/> when <see cref="Guard.IsNull{T}(T,string)"/> (where <typeparamref name="T"/> is <see langword="class"/>) fails.
+        /// </summary>
+        /// <typeparam name="T">The type of the input value.</typeparam>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [DoesNotReturn]
+        internal static void ThrowArgumentExceptionForIsNull<T>(T value, string name)
```

Comment:
name parameter is not documented.
Why did the CI passed?

Reply:
Apparently the current stylecop configuration only complains about missing summaries and type parameter docs when the method is not public, but it's fine with the skipped docs for arguments. I mean, I'm not really complaining since all these methods are just internal anyway, and this lets us save some extra verbosity.

Let me know if it's fine or if you want to update stylecop and add these missing docs in the internal methods too! 👍

Reply:
I just hate incomplete docs xD (hate writing them though)
Its fine, its internal.